x/sysproxy/sysproxy_darwin.go

```diff
@@ -0,0 +1,295 @@
+// Copyright 2024 Jigsaw Operations LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//go:build darwin && !ios
+
+package sysproxy
+
+import (
+	"fmt"
+	"os/exec"
+	"regexp"
+	"strings"
+)
+
+type ProxyType string
+
+const (
+	proxyTypeHTTP  ProxyType = "web"
+	proxyTypeHTTPS ProxyType = "secureweb"
+	proxyTypeSOCKS ProxyType = "socks"
+)
+
+type proxySettings struct {
+	host string
+	port string
+}
+
+func SetWebProxy(host string, port string) error {
+	// Get the active network interface
+	activeInterface, err := getActiveNetworkInterface()
+	if err != nil {
+		return err
+	}
+
+	// Set the web proxy and secure web proxy
+	if err := setProxySettings(proxyTypeHTTP, activeInterface, host, port); err != nil {
+		return err
+	}
+	if err := setProxySettings(proxyTypeHTTPS, activeInterface, host, port); err != nil {
+		// revert previous changes
+		return err
+	}
+
+	return nil
+}
+
+func ClearWebProxy() error {
+	// Get the active network interface
+	activeInterface, err := getActiveNetworkInterface()
+	if err != nil {
+		return err
+	}
+
+	// Clear proxy settings
+	if err := setProxySettings(proxyTypeHTTP, activeInterface, "127.0.0.1", "0"); err != nil {
+		return err
+	}
+	if err := setProxySettings(proxyTypeHTTPS, activeInterface, "127.0.0.1", "0"); err != nil {
+		return err
+	}
+
+	// Unset the web proxy and secure web proxy
+	if err := disableProxy(proxyTypeHTTP, activeInterface); err != nil {
```

Comment:
Disable first, then clear the proxy. Otherwise someone may try to use it with the bogus address. Same for the other platforms.

Reply:
@fortuna I was doing that before but setting proxy parameters automatically change the proxy state to manual and turn it on. I had to change the order otherwise the proxy stayed on.

Reply:
I think I saw this odd behavior on Linux, we might be able to disable proxy on Windows and MacOS before clearing the settings. Need to investigate this again...

Reply:
@fortuna I confirm both on MacOS and Linux changing the proxy server and port automatically turns on the proxy. On Windows, we have to explicitly enable and disable the proxy settings for changes to take effect. This means we must clear the settings before disabling the proxy - the reverse order would leave the proxy state ON.  

If we do not clear / change the setting, we can just disable the proxy.

Reply:
Thanks for testing. If we can't clear the settings, perhaps we just disable it then?

We would have to rename the method to DisableXPRoxy()